### PR TITLE
checkout: also chmod in the user checkout case

### PR DESCRIPTION
**Pull request by jlebon**
When falling back to copying, we previously would only chmod checked out
files in the non-user-checkout mode. Fix this by always doing chmod.
The file_mode was being prepared but never actually applied.

Add a basic test in the archive-z2 --> usermode checkout case in which
we're guaranteed to always fall back to copy mode.

Closes: #633

### CODE REVIEW

**Comment by cgwalters** [comment]
Nice, thanks!

@rh-atomic-bot r+ 618091c

**Comment by jlebon** [comment]
OK, hopefully this works now!

File: tests/basic-test.sh [hunk 1]
```diff
@@ -557,8 +557,13 @@ ${CMD_PREFIX} ostree --repo=repo2 pull-local repo
 rm -rf test2-checkout
 ${CMD_PREFIX} ostree --repo=repo2 checkout -U --disable-cache test2 test2-checkout
 if test -d repo2/uncompressed-objects-cache; then
-    assert_file_has_mode test2-checkout/baz/cowro 600
-    assert_file_has_mode test2-checkout/baz/deeper/ohyeahx 755
+    # we're in archive mode, but the repo we pull-local from might be
```
Comment (cgwalters) [inline tests/basic-test.sh]: Why is this underneath the `if test -d repo2/uncompressed-objects-cache; then` conditional?

**Comment by cgwalters** [comment]
Ah yes, bare-user-only.  Thanks for the docs patch too.  Just one question I noticed, otherwise LGTM (still).

File: tests/basic-test.sh (hunk 1 above)
Comment (jlebon) [inline tests/basic-test.sh]: No good reason at all. :) I had somehow missed the fact this was an error path. I did push another commit to make this more obvious. ⬇️

**Comment by cgwalters** [comment]
@rh-atomic-bot r+  e1c7336

**Pull request closed without merging**